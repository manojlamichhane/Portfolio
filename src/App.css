.App {
  /* text-align: center; */
  background-color: #f8f9fa;
}

.frame {
  width: "80px";
  height: "100px";
}

.active {
  font-weight: "bold";
  color: #fbbe5c;
}
.contactIcon {
  width: 45px;
  height: 45px;
  border-radius: 25px;
  background-color: #fbbe5c;
  padding: 10px;
}
