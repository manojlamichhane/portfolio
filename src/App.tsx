import { useState } from "react";
import { SelectedPage } from "./constants/type";
import {
  Navbar,
  AboutMe,
  Skills,
  Projects,
  Experience,
  ContactMe,
  Footer,
} from "./components";

function App() {
  const [selectedPage, setSelectedPage] = useState<SelectedPage>(
    SelectedPage.AboutMe
  );

  const scrollHandler = (value: SelectedPage) => {
    setSelectedPage(value);
    const nextSection = document.getElementById(value);
    if (nextSection) {
      nextSection.scrollIntoView({ behavior: "smooth" });
    }
  };

  return (
    <div className="bg-base-200">
      <Navbar selectedPage={selectedPage} setSelectedPage={scrollHandler} />
      <AboutMe setSelectedPage={setSelectedPage} />
      <Skills setSelectedPage={setSelectedPage} />
      <Projects setSelectedPage={setSelectedPage} />
      <Experience setSelectedPage={setSelectedPage} />
      <ContactMe setSelectedPage={setSelectedPage} />
      <Footer />
    </div>
  );
}

export default App;
