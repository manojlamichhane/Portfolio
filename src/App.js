import "./App.css";
import Contact from "./components/Contact";
import Home from "./components/Home";
import NavigationBar from "./components/NavigationBar";
import Projects from "./components/Projects";
import Skills from "./components/Skills";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";

function App() {
  return (
    <Router>
      <div className="App">
        <NavigationBar />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/Skills" component={Skills} />
          <Route path="/Projects" component={Projects} />
          <Route path="/Contact" component={Contact} />
        </Switch>
      </div>
    </Router>
  );
}

export default App;
