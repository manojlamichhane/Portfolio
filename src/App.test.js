import { render, cleanup, fireEvent } from "@testing-library/react";
import App from "./App";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

const renderWithRouter = (component) => {
  const history = createMemoryHistory();
  return {
    ...render(<Router history={history}>{component}</Router>),
  };
};

afterEach(cleanup);
test("take a snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});

test("should render home page", () => {
  const { container, getByTestId } = renderWithRouter(<App />);
  const navbar = getByTestId("navbar");
  const link = getByTestId("home-link");

  fireEvent.click(link);
  expect(container.innerHTML).toMatch("Manoj");
  expect(navbar).toContainElement(link);
});
test("should render skills page", () => {
  const { container, getByTestId } = renderWithRouter(<App />);
  const navbar = getByTestId("navbar");
  const slink = getByTestId("skills-link");

  fireEvent.click(slink);
  expect(container.innerHTML).toMatch("Skills");
  expect(navbar).toContainElement(slink);
});
test("should render projects page", () => {
  const { container, getByTestId } = renderWithRouter(<App />);
  const navbar = getByTestId("navbar");
  const plink = getByTestId("projects-link");

  fireEvent.click(plink);
  expect(container.innerHTML).toMatch("projects");
  expect(navbar).toContainElement(plink);
});
test("should render contact page", () => {
  const { container, getByTestId } = renderWithRouter(<App />);
  const navbar = getByTestId("navbar");
  const clink = getByTestId("contact-link");

  fireEvent.click(clink);
  expect(navbar).toContainElement(clink);
  expect(container.innerHTML).toMatch("Contact");
});
