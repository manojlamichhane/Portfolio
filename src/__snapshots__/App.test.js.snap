// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="navbar navbar-expand-md navbar-light bg-light"
      style="justify-content: flex-end;"
    >
      <ul
        class="navbar-nav"
        data-testid="navbar"
      >
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="nav-link active"
            data-testid="home-link"
            href="/"
            style="font-weight: bold; color: rgb(251, 190, 92);"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="skills-link"
            href="/Skills"
          >
            Skills
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="projects-link"
            href="/Projects"
          >
            Projects
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="contact-link"
            href="/Contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <div
      data-testid="home"
      style="background-color: rgb(248, 249, 250); text-align: center;"
    >
      <h1>
        Hi I Am
      </h1>
      <h1
        style="color: rgb(251, 190, 92);"
      >
        Manoj Lamichhane
      </h1>
      <div
        style="display: flex; justify-content: space-between; margin-top: 100px;"
      >
        <p
          style="width: 65%; background-color: white; padding: 20px; border-radius: 50px;"
        >
          I would express myself as a passionate and industrious student with an appetite for learning and enhancing new technical abilities to corporate with the rapidly changing technical world. Pushing myself to the absolute limit, I wish to find the solutions of problem I face while modeling the real work scenario. Wishing to find the bridge between the academic and job based learning so that I can improve and be a perfect professional.
        </p>
        <p
          style="width: 35%; background-color: rgb(251, 190, 92); padding: 20px; border-radius: 50px;"
        >
          To start from the scratch on how one becomes a professional engineer after completing academic education with the objectives of the organization always in mind and perform the tasks allocated without any deviation within the specified time and budget.
        </p>
      </div>
      <div>
        <a
          href="/Contact"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained"
            style="margin-top: 100px; margin-bottom: 60px; background-color: rgb(251, 190, 92); width: 200px; height: 50px; border-radius: 25px;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Contact Me
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
